plugins/browser-plugin-media-tracking/src/api.ts

```diff
@@ -0,0 +1,70 @@
+import { Logger } from '@snowplow/tracker-core';
+import { BrowserPlugin, BrowserTracker } from '@snowplow/browser-tracker-core';
+import { waitForElement } from './findElem';
+import { Config, isElementConfig, isStringConfig } from './config';
+import { setUpListeners } from './player';
+import { setConfigDefaults } from './helperFunctions';
+
+// These imports are used for documentation purposes only.
+// Typescript complains that they are unused.
+// @ts-ignore: TS6133
+import { DynamicContext } from '@snowplow/tracker-core';
+// @ts-ignore: TS6133
+import { HTML5MediaEventTypes } from './config';
+// @ts-ignore: TS6133
+import { FilterOutRepeatedEvents } from '@snowplow/browser-plugin-media/src/types';
+
+let LOG: Logger;
+const _trackers: Record<string, BrowserTracker> = {};
+
+export function MediaTrackingPlugin(): BrowserPlugin {
+  return {
+    activateBrowserPlugin: (tracker: BrowserTracker) => {
+      _trackers[tracker.id] = tracker;
+    },
+    logger: (logger) => {
+      LOG = logger;
+    },
+  };
+}
+
+/**
+ * Enables media tracking on an HTML `<video>` or `<audio>` element.
+ *
+ * @param {Config} config - Configuration options for media tracking.
+ *
+ * **Required:**
+ * - `config.video` (string | HTMLMediaElement): The ID of the media element or the element itself.
+ *
+ * **Optional:**
+ * - `config.label` (`string`): A human-readable label for the media element.
+ * - `config.captureEvents` {@link HTML5MediaEventTypes}: A list of media events to track. All events are tracked by default.
+ * - `config.boundaries` (`number[]`): Percentage thresholds (0-100) at which to trigger progress events. Disabled by default.
+ * - `config.context` {@link DynamicContext}: Contexts to attach to each tracking event.
+ *
+ * **Ping Configuration (Optional):**
+ * - `config.pings.pingInterval` (`number`): The interval (in seconds) for sending ping events. Default is 30(s).
+ * - `config.pings.maxPausedPings` (`number`): The maximum number of ping events sent while playback is paused. Default is 1.
+ *
+ * **Other Options (Optional):**
+ * - `config.updatePageActivityWhilePlaying` (`boolean`): Whether to update page activity while media is playing. Enabled by default.
+ * - `config.filterOutRepeatedEvents` {@link FilterOutRepeatedEvents}: Whether to suppress consecutive identical events. Default is false.
+ */
+export function enableMediaTracking(config: Config) {
```

Comment:
I think we will need a `disableMediaTracking` function as well – that one should unsubscribe the listeners and call `endMediaTracking` on the underlying media package.

Reply:
Come to think of it, do you think we should rename the functions to `startMediaTracking` and `endMediaTracking` to reflect the Vimeo and media plugins?

If we keep it `enableMediaTracking` and add the `id` property (for the media session id), this will be rather tricky for users to upgrade because we will be changing the meaning of the `id` property – previously it identified the video element, now the media session. So I'd prefer if we change the function completely to make users' apps break rather than fail silently.

Reply:
That sounds like a sensible idea to me.